pyseq.py

```diff
@@ -879,3 +908,47 @@ def get_sequences(source):
     log.debug('time: %s' % (datetime.now() - start))
 
     return list(seqs)
+
+
+def walk(source, level=-1, topdown=True, onerror=None, followlinks=False):
+    """ Generator that traverses a directory structure starting at
+        source looking for sequences.
+        :param source: valid folder path to traverse
+        :param level: int, if < 0 traverse entire structure otherwise
+                      traverse to given depth
+        :param topdown: walk from the top down
+        :param onerror: callable to handle os.listdir errors
+        :param followlinks: whether to follow links
+    """
+    start = datetime.now()
+    assert isinstance(source, basestring) is True
+    assert os.path.exists(source) is True
+    source = os.path.abspath(source)
+    for root, dirs, files in os.walk(source, topdown, onerror, followlinks):
+        if topdown is True:
+            parts = root.replace(source, "").split(os.sep)
+            while "" in parts:
+                parts.remove("")
+            if len(parts) == level - 1:
+                del dirs[:]
+
+        files.sort(key=_natural_key)
+        seqs = []
```

Comment:
OK, sounds good. For now, I'm going to check in my fix that uses get_sequences. If you find the bug, feel free to submit another PR with some timing tests. Otherwise, I think it makes sense to just use get_sequences because it should work with groups also, and maybe we could just concentrate on making get_sequences faster.